src/Sylius/Behat/Context/Ui/Shop/CartContext.php

```diff
@@ -270,52 +271,98 @@ public function iAddProductsToTheCart(ProductInterface $product, $quantity)
     }
 
     /**
-     * @Then I should see :elementName
+     * @Then I should be on my cart summary page
      */
-    public function iShouldSeeElement($elementName)
+    public function shouldBeOnMyCartSummaryPage()
     {
         Assert::true(
-            $this->summaryPage->isElementOnPage($elementName),
-            sprintf('Element %s should appear on the page, but it does not.', $elementName)
-         );
+            $this->summaryPage->isOpen(),
+            'Cart summary page should be open, but it does not.'
+        );
     }
 
     /**
-     * @Given unit price value should be :price
+     * @Then I should be notified that the product has been successfully added
      */
-    public function unitPriceValueShouldBe($price)
+    public function iShouldBeNotifiedThatItHasBeenSuccessfullyAdded()
     {
-        $this->summaryPage->open();
-        Assert::eq(
-            $this->summaryPage->getUnitPrice(),
-            $price,
-            'Unit price should be %2$s, but it is %s'
+        $this->notificationChecker->checkNotification('Item has been added to cart.', NotificationType::success());
+    }
+
+    /**
+     * @Then there should be one item in my cart
+     */
+    public function thereShouldBeOneItemInMyCart()
+    {
+        Assert::true(
+            $this->summaryPage->isSingleItemOnPage(),
+            'There should be only one item on list, but it does not.'
         );
     }
 
     /**
-     * @Given total value should be :total
+     * @Then this item should have name :itemName
      */
-    public function totalValueShouldBe($total)
+    public function thisProductShouldHaveName($itemName)
     {
-        $this->summaryPage->open();
-        Assert::eq(
-            $this->summaryPage->getTotal(),
-            $total,
-            'Total should be %2$s, but it is %s'
+        Assert::true(
+            $this->summaryPage->isItemWithName($itemName),
+            sprintf('The product with name %s should appear on the list, but it does not.', $itemName)
         );
     }
 
     /**
-     * @Given quantity value should be :quantity
+     * @Then this item should have variant :variantName
      */
-    public function quantityValueShouldBe($quantity)
+    public function thisItemShouldHaveVariant($variantName)
     {
-        $this->summaryPage->open();
-        Assert::eq(
-            $this->summaryPage->getQuantity(),
+        Assert::true(
+            $this->summaryPage->isItemWithVariant($variantName),
+            sprintf('The product with variant %s should appear on the list, but it does not.', $variantName)
+        );
+    }
+
+    /**
+     * @When /^I add (this product) with ([^"]+) "([^"]+)" to the cart$/
+     */
+    public function iAddThisProductWithSizeToTheCart(ProductInterface $product, $optionName, $optionValue)
+    {
+        $this->productShowPage->open(['slug' => $product->getSlug()]);
+
+        $option = $this->sharedStorage->get(sprintf('%s_option', $optionName));
```

Comment:
You should create ``ProductOptionContext`` transformer, with ``getProductOption()`` or something like that.